app/controllers/application_controller.rb

```diff
@@ -53,6 +52,7 @@ def handle_unverified_request
   after_action :add_noindex_header_to_non_canonical, if: :spa_boot_request?
   after_action :set_cross_origin_opener_policy_header, if: :spa_boot_request?
   after_action :clean_xml, if: :is_feed_response?
+  after_action :set_current_user_for_logs
```

Comment:
Does `Logster.add_to_env()` still work when called in `after_action`?

Reply:
Also... are `after_action` hooks called when an exception is thrown? This username header is important for debugging, so we need to make sure it's present for server-error responses.